test/ServerTests/ServerHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Server.Testing;
+using Microsoft.Extensions.Logging;
+
+namespace ServerTests
+{
+    public static class ServerHelper
+    {
+        private static string GetPathToApplication(string applicationName)
+        {
+            return Path.GetFullPath(Path.Combine("..", "..", "samples", applicationName));
+        }
+
+        public static async Task ServerResponseMessageTest(string siteName, ServerType serverType, RuntimeFlavor runtimeFlavor, RuntimeArchitecture architecture, string applicationBaseUrl,
+            Func<HttpClient, ILogger, CancellationToken, Task> validator)
+        {
+            var logger = new LoggerFactory()
+                            .AddConsole()
+                            .CreateLogger(string.Format("ResponseFormats:{0}:{1}:{2}", serverType, runtimeFlavor, architecture));
+
+            using (logger.BeginScope("ResponseFormatsTest"))
+            {
+                var deploymentParameters = new DeploymentParameters(GetPathToApplication(siteName), serverType, runtimeFlavor, architecture)
+                {
+                    ApplicationBaseUriHint = applicationBaseUrl,
+                    Command = serverType == ServerType.WebListener ? "web" : "kestrel",
+                    SiteName = "HttpTestSite",
+                };
+
+                using (var deployer = ApplicationDeployerFactory.Create(deploymentParameters, logger))
+                {
+                    var deploymentResult = deployer.Deploy();
+                    var httpClientHandler = new HttpClientHandler();
+                    var httpClient = new HttpClient(httpClientHandler) { BaseAddress = new Uri(deploymentResult.ApplicationBaseUri) };
```

Comment:
Set the timeout to something like 5 seconds: https://github.com/aspnet/IISIntegration/blob/dev/test/Microsoft.AspNet.IISPlatformHandler.FunctionalTests/NtlmAuthentationTest.cs#L49  Otherwise it may wait 100 seconds.